stacker/actions/build.py

```diff
@@ -53,119 +60,112 @@ def should_submit(stack):
+    if final_missing:
+        raise MissingParameterException(final_missing)
+
+    return params.items()
+
+
 class Action(BaseAction):
     """Responsible for building & coordinating CloudFormation stacks.
 
     Generates the build plan based on stack dependencies (these dependencies
-    are determined automatically based on references to output values from
-    other stacks).
+    are determined automatically based on output lookups from other stacks).
 
     The plan can then either be printed out as an outline or executed. If
     executed, each stack will get launched in order which entails:
 
         - Pushing the generated CloudFormation template to S3 if it has changed
         - Submitting either a build or update of the given stack to the
-          `Provider`.
-        - Stores the stack outputs for reference by other stacks.
+            :class:`stacker.provider.base.Provider`.
 
     """
 
-    def _resolve_parameters(self, parameters, blueprint):
-        """Resolves parameters for a given blueprint.
-
-        Given a list of parameters, first discard any parameters that the
-        blueprint does not use. Then, if a parameter is a list of outputs
-        in the format of <stack_name>::<output_name>,... pull those output(s)
-        from the foreign stack(s).
+    def build_parameters(self, stack, provider_stack):
+        """Builds the CloudFormation Parameters for our stack.
 
         Args:
-            parameters (dict): A dictionary of parameters provided by the
-                stack definition
-            blueprint (:class:`stacker.blueprint.base.Blueprint`): A Blueprint
-                object that is having the parameters applied to it.
-
-        Returns:
-            dict: The resolved parameters.
-
-        """
-        return resolve_parameters(parameters, blueprint, self.context,
-                                  self.provider)
-
-    def build_parameters(self, stack, provider_stack=None):
-        """Builds the parameters for our stack
-
-        Args:
-            stack (:class:`cloudformation.stack`): A Cloudformation stack
-            provider_stack (:class:`stacker.providers.base.Provider`): An
-                optional Stacker provider object
+            stack (:class:`stacker.stack.Stack`): A stacker stack
+            provider_stack (dict): An optional Stacker provider object
 
         Returns:
             dict: The parameters for the given stack
+
         """
-        parameters = self._resolve_parameters(stack.cfn_parameters,
-                                              stack.blueprint)
-        required_params = [k for k, v in stack.blueprint.required_parameters]
-        parameters = self._handle_missing_parameters(parameters,
-                                                     required_params,
-                                                     provider_stack)
+        resolved = _resolve_parameters(stack.parameters, stack.blueprint)
```

Comment:
We talked about this, but figured I'd put this here - I think we should start referring to `parameters` in this usage instead as `parameter_values` to be more explicit, and help avoid confusion going forward.